### PR TITLE
Update docs for validation errors

### PR DESCRIPTION
**Pull request by dmontagu**
* Review and update `validation_errors.md`
* Fix a couple type errors in the conversion_table.py (noticed while trying to add examples to the `validation_errors.md`)
* Replace the direct use of the schema classes in GenerateSchema with schema functions for consistency with other schema creation (also noticed while trying to add examples to the `validation_errors.md`)
* Add a test that ensures that all error codes in `pydantic_core.core_schema.ErrorType` are either documented or manually removed from the expected values (this is done when the error cannot be produced by code paths currently exposed in pydantic, rather than pydantic-core)

This was brutal to go through, I think I got everything but wouldn't be surprised if there was still a couple minor inconsistencies or typos.

Selected Reviewer: @hramezani

### CODE REVIEW

**Comment by dmontagu** [comment]
please review

File: docs/usage/validation_errors.md [hunk 1]
```diff
@@ -123,13 +147,16 @@ class Model(BaseModel):
...
+    print(repr(exc.errors()[0]['type']))
+    #> 'bytes_type'
 ```
 
+This error is also raised for strict fields when the input value is not an instance of `bytes`.
+
 ## `callable_type`
 
-This error is raised when the value is not a `Callable`.
+This error is raised when the input value is valid as a `Callable`:
```
Comment (hramezani) [inline docs/usage/validation_errors.md:159]:
```suggestion
This error is raised when the input value is not valid as a `Callable`:
```

File: docs/usage/validation_errors.md [hunk 2]
```diff
@@ -380,34 +353,14 @@ class Model(BaseModel):
 
 try:
     Model(x=datetime(2000, 1, 1))
-except ValidationError as exc_info:
-    assert exc_info.errors()[0]['type'] == 'datetime_future'
-```
-
-## `datetime_naive`
```
Comment (hramezani) [inline docs/usage/validation_errors.md:387]: Why did you remove this error?

File: docs/usage/validation_errors.md [hunk 3]
```diff
@@ -792,13 +754,48 @@ class Model(BaseModel):
...
+    print(repr(exc.errors()[0]['type']))
+    #> 'int_parsing_size'
+
+# from JSON
+i64_max = 9_223_372_036_854_775_807
+try:
+    Model.model_validate_json(json.dumps(-i64_max * 2))
+except ValidationError as exc:
+    print(repr(exc.errors()[0]['type']))
+    #> 'dict_type'
```
Comment (hramezani) [inline docs/usage/validation_errors.md:793]: we are in the `int_parsing_size` section, but the error here is `dict_type`

File: docs/usage/validation_errors.md [hunk 4]
```diff
@@ -1069,14 +1081,15 @@ class Model(BaseModel):
 
 
 try:
-    Model(x=BadMapping())
-except ValidationError as exc_info:
-    assert exc_info.errors()[0]['type'] == 'mapping_type'
+    Model(x=None)
+except ValidationError as exc:
+    print(repr(exc.errors()[0]['type']))
+    #> 'dict_type'
```
Comment (hramezani) [inline docs/usage/validation_errors.md:1087]: Shouldn't it be `mapping_type`?

File: docs/usage/validation_errors.md [hunk 5]
```diff
@@ -1165,29 +1180,18 @@ class Model(BaseModel):
...
-from pydantic import BaseModel
-
-
-class Model(BaseModel):
-    x: str
-
-
-Model.model_validate(Model(x='test'), strict=True)
+except ValidationError as exc:
+    print(repr(exc.errors()[0]['type']))
```
Comment (hramezani) [inline docs/usage/validation_errors.md:1188]: I think you forgot to add `#> model_class_type ` as `print` statement output.
I am curious how the test passes without this.

File: docs/usage/validation_errors.md [hunk 6]
```diff
@@ -1494,13 +1516,59 @@ class Model(BaseModel):
...
+
+
+class Model(BaseModel):
+    x: NaiveDatetime
+
+
+try:
+    Model(x=datetime.now(tz=timezone.utc))
+except ValidationError as exc:
+    print(repr(exc.errors()[0]['type']))
```
Comment (hramezani) [inline docs/usage/validation_errors.md:1566]: same here. print statement output is missed

**Comment by hramezani** [comment]
Great work @dmontagu 

I left some comments.

Please update

File: docs/usage/validation_errors.md (hunk 3 above)
Comment (dmontagu) [reply docs/usage/validation_errors.md:793]: good catch, I'll fix it

File: docs/usage/validation_errors.md (hunk 4 above)
Comment (dmontagu) [reply docs/usage/validation_errors.md:1087]: yep need to fix that too

File: docs/usage/validation_errors.md (hunk 5 above)
Comment (dmontagu) [reply docs/usage/validation_errors.md:1188]: it means it's not hitting the error, which is why I changed to `print` from the way it was before — the old way, since it did an assert, you couldn't tell if it was hitting the error. Good catch, I'll fix this.

File: docs/usage/validation_errors.md (hunk 6 above)
Comment (dmontagu) [reply docs/usage/validation_errors.md:1566]: thanks

**Comment by dmontagu** [comment]
@hramezani I think I addressed everything you pointed out. Please review
